packages/playwright/src/reporter.ts

```diff
@@ -288,14 +296,20 @@ class ReplayPlaywrightReporter implements Reporter {
         }
       }
 
-      steps.push(step);
+      if (hook) {
+        const hookSteps = hookMap.get(hook) || [];
+        hookSteps.push(step);
+        hookMap.set(hook, hookSteps);
+      } else {
+        steps.push(step);
+      }
     });
 
     return {
       beforeAll: [],
       afterAll: [],
```

Comment:
Those are not supported right now. I hope most users don't use those - you can't even rely on `page` there:
```
    Error: "context" and "page" fixtures are not supported in "beforeAll" since they are created on a per-test basis.
    If you would like to reuse a single page between tests, create context manually with browser.newContext(). See https://aka.ms/playwright/reuse-page for details.
    If you would like to configure your page before each test, do that in beforeEach hook instead.

    Error: "context" and "page" fixtures are not supported in "afterAll" since they are created on a per-test basis.
    If you would like to reuse a single page between tests, create context manually with browser.newContext(). See https://aka.ms/playwright/reuse-page for details.
    If you would like to configure your page before each test, do that in beforeEach hook instead.
```

I played with this:
```ts
let externalPage: any;

test.beforeAll(async ({ browser }) => {
  externalPage = await browser.newPage();
  await externalPage.goto("http://localhost:3000");
  await externalPage
    .evaluate(() => {
      console.log("beforeAll!");
    })
    .catch(() => {});
});

test.afterAll(async () => {
  await externalPage
    .evaluate(() => {
      console.log("afterAll!");
    })
    .catch(() => {});
  await externalPage.close();
});
```

But then I get more than one recording... and it's somewhat confusing to reason about. I don't think this problem is specific to those hooks though - it's a more general UX problem for tests that create multiple pages (and even we have those in our e2e tests). I need to dig deeper into this later to understand how exactly our data is structured and how it relates to what we show on specific recordings.

Reply:
Thanks for exploring. I'll make a ticket to chase this topic down in a future milestone.